lib/kontena/templates/haproxy/http_in.text.erb

```diff
@@ -0,0 +1,41 @@
+listen http-in
+    bind *:80
+    http-request replace-header Host (.*):.* \1
+    <% if ENV['SSL_CERTS'] %>
+    bind *:443 ssl crt /etc/haproxy/certs/ no-sslv3
+    reqadd X-Forwarded-Proto:\ https if { ssl_fc }
+    reqadd X-Forwarded-Port:\ 443 if { ssl_fc }
+    <% end %>
+
+    errorfile 502 /etc/haproxy/errors/502.http
+    errorfile 503 /etc/haproxy/errors/503.http
+
+    <% if ENV['KONTENA_LB_HEALTH_URI'] %>
+    monitor-uri <%= ENV['KONTENA_LB_HEALTH_URI'] %>
+    errorfile 200 /etc/haproxy/errors/200.http
+    <% end %>
+    
+<% services.each do |service| %>
+<% if service.upstreams? %>
```

Comment:
This is different vs the equivalent conditional in the current confd templates, because those are actually buggy: #26

Reply:
Verified that this also "fixes" the tests in #26, and I'm not sure that's a good thing... modifying the tests to expect the service-b definition to still be there, it's now gone. The current `kontena/lb:latest` code will keep the `use_backend service-b` even if it doesn't have any upstreams:

#### `test/no_upstreams_test.bats`
```sh
#!/usr/bin/env bats

load "common"


setup() {
  etcdctl rm --recursive /kontena/haproxy/lb/services/service-b || true
}


@test "if no upstreams, service frontend retained in config" {
  etcdctl set /kontena/haproxy/lb/services/service-b/virtual_hosts www.foo.com

  sleep 1
  run curl -sL -w "%{http_code}" -H "Host: www.foo.com" http://localhost:8180/ -o /dev/null
  [ "${lines[0]}" = "503" ]

  run config
  assert_output_contains "use_backend service-b" 1
}
```

#### `bats test/no_upstreams_test.bats` (master)
```
 ✓ if no upstreams, service omitted from config
```

#### `bats test/no_upstreams_test.bats` (feature/remove-confd-and-rsyslog)
```
 ✗ if no upstreams, service frontend retained in config
   (from function `assert_equal' in file test/common.bash, line 29,
    from function `assert_output_contains' in file test/common.bash, line 39,
    in test file test/no_upstreams_test.bats, line 19)
     `assert_output_contains "use_backend service-b" 1' failed
     global
   www.foo.com
   expected: 1
   actual:   0
```

Reply:
See #29 for a test for this "bug"

Reply:
Maybe we need to revert changes related to that bug fix.

Reply:
AFAIK just removing the `<% if service.upstreams? %>` conditional should be enough to make this pass #29